Tests/OAuth/ResourceOwner/AzureResourceOwner.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the HWIOAuthBundle package.
+ *
+ * (c) Hardware.Info <opensource@hardware.info>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace HWI\Bundle\OAuthBundle\Tests\OAuth\ResourceOwner;
+
+use HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\AzureResourceOwner;
+
+class AzureResourceOwnerTest extends GenericOAuth2ResourceOwnerTest
+{
+    protected $csrf = true;
+
+    protected $userResponse = <<<json
+{
+    "id":  "1",
+    "foo": "bar"
```

Comment:
Yes, that was the aim.
But in order to do this for this Azure resource, you need to pass an encoded id_token (not the access token) that gets decoded in some way. I was not able to create such a fake token that gets decoded correctly, so validating the user data parsing doesn't work (except with a real token which I tested, but did not want toe submit my credentials here).

I'd like to change it if you really want it but what exactly do you want done?